crates/saya/core/src/lib.rs

```diff
@@ -51,25 +59,19 @@ pub struct SayaConfig {
     #[serde(deserialize_with = "url_deserializer")]
     pub prover_url: Url,
     pub prover_key: ProverAccessKey,
+    pub mode: SayaMode,
+    pub piltover_contract: Felt,
```

Comment:
More than piltover, this should be settlement contract as piltover is already very specific.

Also @neotheprogramist do you remember we mentioned that we may want to have a derived contract for piltover to actually permit users to settle their app chain on already deployed contract without having them to actually do all the maintenance and work to deploy piltover?

In this case, one should be able to register a new app chain on the piltover-multi contract. When registered, they can modify the settings (config, account address allowed to push update etc...), and then it's done. They will gain for us maintaining this core contract when multiple dojo app-chains can live.

We can catch up on that, but would be great if we could have this to reduce the friction using saya for Dojo devs.

@matzayonc if it's sound clear to you, I can fork piltover under dojo and you could work on that.

Reply:
@glihm Do you think we should inspire by dojo architecture and have piltover contract as `storage` and other contracts for app chains?

Reply:
Seems straightforward enough to implement multiple appchain support. Because we provide the `class-hash` in the README, this would just be one `snacast` call less. We could also register appchain from `saya` itself if not provided.

Reply:
> k we should inspire by dojo architecture and have

Sorry could you rephrase, not sure I get you on this one.